### ISSUE TITLE
Fluid temperature is broken

### ISSUE THREAD

**Issue by ClaudeMetz** [ClaudeMetz/FactoryPlanner#95]
The current system barely works, but is not extensible to make it fully work. This needs a 'throw everything out and try again'-approach. I need to figure out how all this temperature stuff works in detail before trying to implement a system for it.

Recipes that don‘t specify a specific fluid temperature of an ingredient can accept any temperature as input. This is not respected currently, as it only accepts steam without temperature itself, which is silly. This needs custom logic that tests for this, might be annoying to do in a performant way.

Fluid fuel also cares about temperature apparently, so that also needs to be addressed

This image has some un-investigated theory for how to go about this

![image](https://github.com/ClaudeMetz/FactoryPlanner/assets/22834555/91f6d7e3-2e47-4faf-ad4e-c7b8f69e68a0)

Maybe some relevant discussion https://github.com/ShadowTheAge/yafc/pull/67

**Comment by ClaudeMetz** [comment]
After weeks of work, I think I am close to cracking this. Just didn't make it for 2.0, but will hopefully come soon after.

**Comment by puggan** [comment]
I noticed that the pyAddons recipt for heating stone bricks with high temp coke gas, that return the same amount of low temp coke gas, currently ignores the coke gas in the solver, as it nets ±0 total coke gas.
Is that part of this issue? (or should I write a separate one?)

**Comment by ClaudeMetz** [comment]
Yes this is related to temperature, it cancels them out but it obviously shouldn't.
